fv3core/stencils/xppm.py

```diff
@@ -168,236 +106,247 @@ def xt_dxa_edge_1_base(q, dxa):
+    dm: FloatField,
+    al: FloatField,
+):
+    bl = al - q
+    xt = yppm.s15 * q[1, 0, 0] + yppm.s11 * q + yppm.s14 * dm
+    br = xt - q
+    return bl, br
 
 
-@gtstencil()
-def east_edge_iord8plus_1(q: sd, dxa: sd, dm: sd, bl: sd, br: sd, xt_minmax: bool):
-    with computation(PARALLEL), interval(...):
-        xt = s15 * q + s11 * q[-1, 0, 0] + s14 * dm[-1, 0, 0]
-        bl = xt - q
-        xt = xt_dxa_edge_0(q, dxa, xt_minmax)
-        br = xt - q
+@gtscript.function
+def east_edge_iord8plus_1(
+    q: FloatField,
+    dxa: FloatField,
+    dm: FloatField,
+):
+    xt = yppm.s15 * q + yppm.s11 * q[-1, 0, 0] + yppm.s14 * dm[-1, 0, 0]
+    bl = xt - q
+    xt = xt_dxa_edge_0(q, dxa)
+    br = xt - q
+    return bl, br
 
 
-@gtstencil()
-def east_edge_iord8plus_2(q: sd, dxa: sd, dm: sd, bl: sd, br: sd, xt_minmax: bool):
-    with computation(PARALLEL), interval(...):
-        xt = xt_dxa_edge_1(q, dxa, xt_minmax)
-        bl = xt - q
-        br = s11 * (q[1, 0, 0] - q) - s14 * dm[1, 0, 0]
+@gtscript.function
+def east_edge_iord8plus_2(
+    q: FloatField,
+    dxa: FloatField,
+    dm: FloatField,
+):
+    xt = xt_dxa_edge_1(q, dxa)
+    bl = xt - q
+    br = yppm.s11 * (q[1, 0, 0] - q) - yppm.s14 * dm[1, 0, 0]
+    return bl, br
 
 
-def compute_al(q, dxa, iord, is1, ie3, jfirst, jlast, kstart=0, nk=None):
-    if nk is None:
-        nk = grid().npz - kstart
-    dimensions = q.shape
-    local_origin = (origin[0], origin[1], kstart)
-    al = utils.make_storage_from_shape(dimensions, local_origin)
-    domain_y = (1, dimensions[1], nk)
-    if iord < 8:
-        main_al(
-            q,
-            al,
-            origin=(is1, jfirst, kstart),
-            domain=(ie3 - is1 + 1, jlast - jfirst + 1, nk),
-        )
-        if not grid().nested and spec.namelist.grid_type < 3:
-            if grid().west_edge:
-                al_y_edge_0(
-                    q, dxa, al, origin=(grid().is_ - 1, 0, kstart), domain=domain_y
-                )
-                al_y_edge_1(q, dxa, al, origin=(grid().is_, 0, kstart), domain=domain_y)
-                al_y_edge_2(
-                    q, dxa, al, origin=(grid().is_ + 1, 0, kstart), domain=domain_y
-                )
-            if grid().east_edge:
-                al_y_edge_0(q, dxa, al, origin=(grid().ie, 0, kstart), domain=domain_y)
-                al_y_edge_1(
-                    q, dxa, al, origin=(grid().ie + 1, 0, kstart), domain=domain_y
-                )
-                al_y_edge_2(
-                    q, dxa, al, origin=(grid().ie + 2, 0, kstart), domain=domain_y
-                )
-        if iord < 0:
-            floor_cap(
-                al,
-                0.0,
-                origin=(grid().is_ - 1, jfirst, kstart),
-                domain=(grid().nic + 3, jlast - jfirst + 1, nk),
-            )
-    return al
+@gtscript.function
+def compute_al(q: FloatField, dxa: FloatField):
+    """
+    Interpolate q at interface.
 
+    Inputs:
+        q: Tracer
```

Comment:
I think q is not always a tracer, should probably label it the same as you have for the outer function "Transported scalar"